### ISSUE TITLE
Minitest/TestMethodName cop auto-fixer breaks things. 

### ISSUE THREAD

**Issue by rbuchberger** [rubocop-hq/rubocop-minitest#115]
The Minitest/TestMethodName cop autofixer performs a simple rename of any methods which are not lifecycle methods and which don't begin with `test_`.  This is not a safe thing to do.

Minitest already only runs methods which begin with `test_`. Everything else is likely a helper method (or a mistake). Renaming a helper method breaks everything which uses it, and creates an empty test.

--------

## Expected behavior

Only make safe changes. You could do one of the following:

* Only flag & rename methods which contain assertions.
* Check whether methods are called anywhere else when renaming them. Given that they could be called from nearly anywhere, this might be a challenge.
* Disable the auto-fixer.
* Consider whether this cop needs to exist? As I said, minitest defines tests as methods named `test_`. Is this cop meant to stop people from accidentally writing tests which don't get run, or are inline helper methods, which are specific to that test class, bad practice?

## Actual behavior

The auto-fixer renamed my helper methods, breaking my tests.

## Steps to reproduce the problem

* Define a helper method within a test class. Here's an example:

```ruby

require 'test_helper'
class SomeTest < Minitest::Test
  include TestHelper

  # Define a helper method
  def tested(foo = 'useful default')
    # do stuff with foo, return test result
  end

  def test_something
    # test setup

    # use helper method:
    assert_equal 'correct answer', tested
  end
end

```

* Run `rubocop --autocorrect`, get this:

```ruby

require 'test_helper'
class SomeTest < Minitest::Test
  include TestHelper

  # Name changed here:
  def test_tested(foo = 'useful default')
    # do stuff
  end

  def test_something
    # setup

    # But not here:
    assert_equal 'correct answer', tested
  end
end

```

If you'd like a real-world example, you can use [jekyll_picture_tag](https://github.com/rbuchberger/jekyll_picture_tag). Setup instructions are pretty simple, documented here:

* [setup](https://rbuchberger.github.io/jekyll_picture_tag/devs/contributing/setup.html)
* [Tests & checks](https://rbuchberger.github.io/jekyll_picture_tag/devs/contributing/testing.html)

Basically run the tests, then autocorrect, then run the tests again. Note that currently that cop is disabled under `test/.rubocop.yml`, you'll have to re-enable it to reproduce the issue.

## RuboCop version

```
$ bundle exec rubocop -V
1.7.0 (using Parser 2.7.2.0, rubocop-ast 1.4.0, running on ruby 2.5.8 x86_64-linux)
  - rubocop-minitest 0.10.2
  - rubocop-rake 0.5.1
```

Also, I just want to say thanks. Your project has made maintaining my project much easier, and I really appreciate that!

**Comment by andyw8** [comment]
cc @fatkodima 

I think at the very least this cop should be marked as not safe for auto-correction.
 
> Consider whether this cop needs to exist? As I said, minitest defines tests as methods named test_. Is this cop meant to stop people from accidentally writing tests which don't get run, or are inline helper methods, which are specific to that test class, bad practice?

The former, I've encountered several test suits containing tests which aren't being run due to invalid naming.

**Comment by rbuchberger** [comment]
That makes sense! I've made the same mistake with an entire test file, so I can see how it would happen.

**Comment by koic** [comment]
Thank you for the feedback and discussion ❤️ 

The cop overlooked that the test method takes no arguments. Furthermore, the above patch #116 will not register an offense when it does not have assertion methods. Because of this, it can increase the prevention of false positives for helper methods.

Therefore, before marking it as unsafe auto-correction, I'd like to take a look at this enhanced condition first.

**Comment by koic** [comment]
I have confirmed the case where #116 applies.

## Before

```console
% cd path/to/jekyll_picture_tag
% bundle exec rubocop --only Minitest/TestMethodName
Inspecting 68 files
..............................................CCC............CCCC...

Offenses:

test/unit/instructions/test_config.rb:19:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested
      ^^^^^^
test/unit/instructions/test_preset.rb:15:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested(name = 'default')
      ^^^^^^
test/unit/instructions/test_tag_parser.rb:10:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested(params = 'img.jpg')
      ^^^^^^
test/unit/test_generated_image.rb:23:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested
      ^^^^^^
test/unit/test_generated_image_actual.rb:20:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested(basename = 'rms', ext = 'jpg')
      ^^^^^^
test/unit/test_generated_image_actual.rb:27:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def source_image(basename, ext)
      ^^^^^^^^^^^^
test/unit/test_generated_image_actual.rb:35:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def config
      ^^^^^^
test/unit/test_generated_image_actual.rb:46:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def out_dir
      ^^^^^^^
test/unit/test_generated_image_actual.rb:50:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def site_stub
      ^^^^^^^^^
test/unit/test_generated_image_actual.rb:57:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def make_dest_dir
      ^^^^^^^^^^^^^
test/unit/test_generated_image_missing.rb:23:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested
      ^^^^^^
test/unit/test_img_uri.rb:23:7: C: [Correctable] Minitest/TestMethodName: Test method name should start with test_ prefix.
  def tested(param = 'img.jpg', source_image: false)
      ^^^^^^

68 files inspected, 12 offenses detected, 12 offenses auto-correctable
```


## After

```console
% bundle exec rubocop --only Minitest/TestMethodName
Inspecting 68 files
....................................................................

68 files inspected, no offenses detected
```
